pages/emotion-records.js

```diff
@@ -32,31 +55,52 @@ export default function EmotionRecords({
   onDeleteEmotionEntry,
   toggleHighlight,
 }) {
+  const [shownEntries, SetShownEntries] = useState(emotionEntries);
   const [isHighlighted, setIsHighlighted] = useState(false);
 
   function handleShowHighlighted() {
     setIsHighlighted(!isHighlighted);
   }
+
+  console.log(shownEntries);
+
   return (
     <StyledWrapper>
-      <StyledTitle>
-        Recorded Emotions{" "}
-        <span>
-          <StyledHighlightButton onClick={handleShowHighlighted}>
-            {isHighlighted ? "Show all Entries" : "Show highlighted Entries"}
-          </StyledHighlightButton>
-        </span>
-      </StyledTitle>
-
-      <EmotionRecordsList
-        onDeleteEmotionEntry={onDeleteEmotionEntry}
-        emotionEntries={
-          isHighlighted
-            ? emotionEntries.filter((entry) => entry.isHighlighted)
-            : emotionEntries
-        }
-        toggleHighlight={toggleHighlight}
-      />
+      <StyledPageHeader>
+        <StyledTitle>Recorded Emotions</StyledTitle>
+
+        {emotionEntries.length !== 0 && (
+          <>
+            <SearchBar
+              SetShownEntries={SetShownEntries}
```

Comment:
it's not good practice to send the setter function to another component as a prop. You better should implement a handler function that updates the state in a way that you define. Then send the handler function as a prop. In this way you protect your state from unexpected changes.